main.go

```diff
@@ -197,6 +197,7 @@ var (
 	}
 	slowLinters = []string{"structcheck", "varcheck", "errcheck", "aligncheck", "testify", "test", "interfacer", "unconvert", "deadcode"}
 	sortKeys    = []string{"none", "path", "line", "column", "severity", "message", "linter"}
+	outputKeys  = []string{"console", "json", "checkstyle"}
```

Comment:
Different output modes like console, JSON and checkstyle are all mutually exclusive. Approach with single output flag was chosen to avoid having to deal with `gometalinter --json --check-style` invocations.

Toggling output modes with booleans also creates asymmetry with console output mode - JSON and checkstyle would have flags while console output mode will be triggered by not having any of those two. It is hard to document this type of behavior in flag descriptions. Single output flag on the other hand clearly states that default mode is console, and there are only two other options.

Single output mode flag also provides an easy way to add more output modes without introducing more flags and more invalid flag combinations.

Support for `--json` is left to be backwards compatibility.

Reviewing the pull-request I noticed that `--output` is probably not the best flag name for this case, it should be changed to `--format` not to confuse with output location like console/file/network.

I will refactor the code if you prefer single `--check-style` flag. Just wanted to explain my reasoning behind this before rushing to change code. Waiting for your input.